src/Microsoft.Health.Dicom.Api/Features/Converter/CustomJsonStringEnumConverterImp.cs

```diff
@@ -0,0 +1,40 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Globalization;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using EnsureThat;
+
+namespace Microsoft.Health.Dicom.Api.Features.Converter
+{
+    internal class CustomJsonStringEnumConverterImp<T> : JsonConverter<T>
+        where T : Enum
+    {
+        public override T Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        {
+            EnsureArg.IsNotNull(typeToConvert, nameof(typeToConvert));
+
+            if (reader.TokenType == JsonTokenType.String)
+            {
+                string content = reader.GetString();
+                if (Enum.TryParse(typeToConvert, content, true, out object result))
```

Comment:
```suggestion
                if (Enum.TryParse<T>(content, true, out T result))
```

Unless there is some screwiness between the `enum` type constraint and `Enum.TryParse<T>` which is totally possible.

Reply:
Get this failure if I do that:
![image](https://user-images.githubusercontent.com/67345894/135931894-8714f757-0fc7-432a-934f-9dddfd9ad31e.png)